lib/Sema/TypeCheckAttr.cpp

```diff
@@ -1732,7 +1762,8 @@ void AttributeChecker::visitSpecializeAttr(SpecializeAttr *attr) {
   }
 
   // Check the validity of provided requirements.
-  checkSpecializeAttrRequirements(attr, FD, resolvedRequirements, TC);
+  checkSpecializeAttrRequirements(attr, FD, resolvedRequirements,
+                                  constrainedGenericParams, TC);
 
   // Store converted requirements in the attribute so that they are
```

Comment:
Yeah. I'm not quite sure about. We need to figure out if it is possible. But some of the checks can be moved into ArchetypeBuilder I guess.